sensor/extrinsic_reflector_based_calibrator/scripts/metrics_plotter_node.py

```diff
@@ -0,0 +1,265 @@
+        self.subplot0.set_xlim(0, xlim)
+        self.subplot1.set_xlim(0, xlim)
+        self.subplot2.set_xlim(0, xlim)
+        self.subplot3.set_xlim(0, xlim)
+
+        self.subplot0.set_ylim(0, self.max_ylim0 + 5)
+        self.subplot1.set_ylim(0, self.max_ylim1 + 0.1)
+        self.subplot2.set_ylim(0, self.max_ylim2 + 5)
+        self.subplot3.set_ylim(0, self.max_ylim3 + 0.1)
+
+    def draw_lines(self):
+        self.subplot0.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_cv_distance_error, self.cv_distance_error],
+            self.color_bo,
+        )
+        self.subplot1.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_cv_yaw_error, self.cv_yaw_error],
+            self.color_go,
+        )
+        self.subplot2.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_calibration_distance_error, self.calibration_distance_error],
+            self.color_bo,
+        )
+        self.subplot3.plot(
+            [self.prev_num_of_reflectors, self.num_of_reflectors],
+            [self.prev_calibration_yaw_error, self.calibration_yaw_error],
+            self.color_go,
+        )
+
+        self.anno0 = self.subplot0.annotate(
+            f"{self.cv_distance_error:.2f}",
+            xy=(self.num_of_reflectors, self.cv_distance_error),
+            color=self.color_b,
+        )
+        self.anno1 = self.subplot1.annotate(
+            f"{self.cv_yaw_error:.2f}",
+            xy=(self.num_of_reflectors, self.cv_yaw_error),
+            color=self.color_g,
+        )
+        self.anno2 = self.subplot2.annotate(
+            f"{self.calibration_distance_error:.2f}",
+            xy=(self.num_of_reflectors, self.calibration_distance_error),
+            color=self.color_b,
+        )
+        self.anno3 = self.subplot3.annotate(
+            f"{self.calibration_yaw_error:.2f}",
+            xy=(self.num_of_reflectors, self.calibration_yaw_error),
+            color=self.color_g,
+        )
+        plt.tight_layout()
+        plt.pause(0.1)
+
+    def draw_with_msg(self, msg):
+        self.num_of_reflectors = msg.data[0]
+        # changing from meters to centimeters
+        self.cv_distance_error = msg.data[1] * 100
+        self.cv_yaw_error = msg.data[2]
+        self.calibration_distance_error = msg.data[3] * 100
+        self.calibration_yaw_error = 0 if math.isnan(msg.data[4]) else msg.data[4]
+
+        self.check_if_deleted()
+        self.update_xy_lim()
+        self.remove_annotations()
+        self.draw_lines()
+
+        self.prev_cv_distance_error = self.cv_distance_error
+        self.prev_cv_yaw_error = self.cv_yaw_error
+        self.prev_calibration_distance_error = self.calibration_distance_error
+        self.prev_calibration_yaw_error = self.calibration_yaw_error
+        self.prev_num_of_reflectors = self.num_of_reflectors
+
+
+class MetricsPlotter(Node):
```

Comment:
Since you are using class composition, it makes sense that MetricsPlotter should be named MetricsPlotterNode

Reply:
Commit https://github.com/tier4/CalibrationTools/pull/145/commits/82d940da6f5dfdebeea22fd0649d4526e63a8c7f fix this.